### PR TITLE
Feature lwdev 7161 add auto registration

### PR DESCRIPTION
**Pull request by starkmsu**

### CODE REVIEW

File: client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs [hunk 1]
```diff
@@ -30,33 +33,64 @@ public static class AutoRegistrationInMonitoring
...
                 throw new ArgumentNullException(nameof(configuration));
+
+            string disableAutoRegistrationStr = configuration[_disableAutoRegistrationEnvVarName];
+            if (bool.TryParse(disableAutoRegistrationStr, out bool disableAutoRegistration) && disableAutoRegistration)
+                return;
+
             if (string.IsNullOrWhiteSpace(monitoringServiceUrl))
                 throw new ArgumentException("Argument is empty", nameof(monitoringServiceUrl));
+
             if (log == null)
```
Comment (KonstantinRyazantsev) [inline client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs:45]: better to do preconditions check at the start of the method

File: client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs [hunk 2]
```diff
@@ -30,33 +33,64 @@ public static class AutoRegistrationInMonitoring
...
                         Url = myMonitoringUrl,
                         ServiceName = myMonitoringName,
                     });
-                log.WriteInfo("Auto-registration in monitoring", "", $"Auto-registered in Monitoring with name {myMonitoringName} on {myMonitoringUrl}");
+                log.WriteMonitor("Auto-registration in monitoring", podTag, $"Auto-registered in Monitoring with name {myMonitoringName} on {myMonitoringUrl}");
             }
             catch (Exception ex)
             {
-                log.WriteError("Auto-registration in monitoring", "", ex);
+                log.WriteMonitor("Auto-registration in monitoring", podTag, ex.ToString());
```
Comment (KonstantinRyazantsev) [inline client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs]: Use ToAsyncString()

File: client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs [hunk 3]
```diff
@@ -30,33 +33,64 @@ public static class AutoRegistrationInMonitoring
...
+                {
+                    var monitoringRegistration = await monitoringService.GetService(myMonitoringName);
+                    if (monitoringRegistration.Url == myMonitoringUrl)
+                        return;
+
+                    log.WriteMonitor("Auto-registration in monitoring", podTag, $"There is a registration for {myMonitoringName} in monitoring service!");
+
+                    myMonitoringUrl = _missingEnvVarUrl;
+                    if (string.IsNullOrEmpty(podTag))
+                        podTag = Guid.NewGuid().ToString();
```
Comment (KonstantinRyazantsev) [inline client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs]: Either declare another variable (i.e. podName), or implement this inititalization logic at line 47, where podTag is defined.

File: client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs (hunk 1 above)
Comment (starkmsu) [reply client/MonitoringServiceApiCaller/AutoRegistrationInMonitoring.cs:45]: This is the start of the method. And all the code before this does checks before actual functionality

**Review by KonstantinRyazantsev — APPROVED** [review]

**Pull request merged**